dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1087,7 +1095,7 @@ class IndexConsumer : public ASTConsumer,
             text[i] != '\t' && text[i] != '\n')
           text[i] = '?';
       }
-      recordValue("text", text, true);
+      macromap.insert(make_pair(nameStart, text));
```

Comment:
It seems like this will keep the full text of any macro body in memory forever. I'm not sure how that's going to go on something as big as Firefox, which can barely be compiled in a 7GB VM as is. What do you think?